pages/details/_title.vue

```diff
@@ -3,46 +3,60 @@
     <v-card elevation="2">
       <v-img
         class="white--text align-end"
-        height="200px"
-        :src="singleData.urlToImage"
+        :src="product.urlToImage"
       >
-        <v-card-title>{{ singleData.title }}</v-card-title>
+        <v-card-title>{{ product.title }}</v-card-title>
       </v-img>
 
       <v-card-subtitle class="pb-0">
-        {{ singleData.description }}
+        {{ product.description }}
       </v-card-subtitle>
 
-      <v-card-text class="text--primary">{{ singleData.author }} </v-card-text>
+      <v-card-text class="text--primary">{{ product.author }} </v-card-text>
       <h5 class="date-v">Published date : 20th January 2023</h5>
     </v-card>
   </div>
 </template>
-<script>
+
+<script lang="ts">
 import axios from 'axios'
-export default {
+import { Dictionary } from 'vue-router/types/router'
+import Vue from 'vue'
+// import Vue from 'vue/types/umd'
+import {SingleDataType} from '../../interface/productInterface'
+export default Vue.extend ({
   name: 'details',
   data() {
+    const data: Array<SingleDataType> = []
+    let product: SingleDataType = {
+      description: '',
+      urlToImage: '',
+      title: '',
+      author: ''
+    }
+
     return {
-      data: [],
-      singleData: [],
+      data,
+      product,
     }
   },
-  async mounted() {
+  mounted() {
     this.fetchData(this.$route.params)
   },
   methods: {
-    async fetchData(params) {
-      await axios
+    fetchData(params: Dictionary<String>): void {
+      axios
         .get(
           'https://newsapi.org/v2/everything?q=bitcoin&apiKey=70905943afe0477ab21103fdbb396454'
         )
         .then((res) => (this.data = res.data.articles))
-
-      this.singleData = this.data.find((e) => e.title === params.title)
+        const filterData: SingleDataType | any = this.data.find((e: SingleDataType) => e.title === params.title)
```

Comment:
if you are accepting any here 
then what's the use of give types to filterData?

Reply:
its gonna be either SIngleDataType or undefined based on filter that's why later on to that added if condition

Reply:
Resolved